shuriken/mjolnir/include/mjolnir/MjolnIROps.td

```diff
@@ -179,64 +179,112 @@ def GetArrayOp: MjolnIR_Op<"getarray",
-        In order to store a value in a field we have to represent the field
-        as a Symbol, since there are not global variables. In that case we
-        will obtain the SSA value that will be stored in the field, and we will
-        obtain the name of the field and reference.
+        Stores a value into a field. The field can be either static or instance-based.
+        For instance fields, an object reference must be provided.
 
+        Examples:
         ```mlir
-            MjolnIR.storeField 
+        // Static field store
+        MjolnIR.storefield static %value : i32, MyClass->myStaticField
+
+        // Instance field store
+        MjolnIR.storefield instance %value : i32, (%obj : !MjolnIR.object<MyClass>) MyClass->myField
         ```
     }];
 
     let arguments = (ins
-        DVM_BasicTypes:$value,   // value to store in the Field
-        FlatSymbolRefAttr:$fieldName, // name of Field
-        FlatSymbolRefAttr:$fieldClass, // class of Field
-        UI32Attr: $fieldRef // reference to Field in Dalvik code
+        DVM_BasicTypes:$value,           // value to store in the Field
+        FlatSymbolRefAttr:$fieldName,    // name of Field
+        FlatSymbolRefAttr:$fieldClass,   // class of Field
+        MjolnIR_FieldTypeAttr:$accessType, // type of field access
+        Optional<DVM_BasicTypes>:$instance    // optional instance object for instance fields
     );
 
-    let assemblyFormat = "attr-dict $value`:`type($value)`,` $fieldClass`->`$fieldName `(`$fieldRef`)`";
+    let assemblyFormat = [{
+        $accessType $value `:` type($value) `,` ($instance^ `:` type($instance))? $fieldClass `->` $fieldName attr-dict
+    }];
+
+    let builders = [
+        // Builder for static field stores
+        OpBuilder<(ins "Value":$value, "StringRef":$fieldName, "StringRef":$fieldClass), [{
+            $_state.addOperands(value);
+            $_state.addAttribute("fieldName", FlatSymbolRefAttr::get($_builder.getContext(), fieldName));
+            $_state.addAttribute("fieldClass", FlatSymbolRefAttr::get($_builder.getContext(), fieldClass));
+            $_state.addAttribute("accessType", ::mlir::shuriken::MjolnIR::FieldAccessTypeAttr::get($_builder.getContext(), 
+                ::mlir::shuriken::MjolnIR::FieldAccessType::STATIC));
+        }]>,
+
+        // Builder for instance field stores
+        OpBuilder<(ins "Value":$value, "StringRef":$fieldName, "StringRef":$fieldClass, "Value":$instance), [{
+            $_state.addOperands(value);
+            $_state.addAttribute("fieldName", FlatSymbolRefAttr::get($_builder.getContext(), fieldName));
+            $_state.addAttribute("fieldClass", FlatSymbolRefAttr::get($_builder.getContext(), fieldClass));
+            $_state.addAttribute("accessType", ::mlir::shuriken::MjolnIR::FieldAccessTypeAttr::get($_builder.getContext(), 
+                ::mlir::shuriken::MjolnIR::FieldAccessType::INSTANCE));
+            $_state.addOperands(instance);
+        }]>
+    ];
```

Comment:
I think we should let mlir auto generate the create helper method for this operation so in the future we do less maintenance.

If you're concerned about a non-optional instance appearing with static enum case then maybe we should switch to a boolean `hasVerifier = 1` for the tablegen

Reply:
These builders are used to avoid the programmer concern about creating a `load/store` static or instance. As you can see, the builders contain different parameters, and depending on the builder, a different `FieldAccessType` is used, in this way it is easier also for us.